src/schematools/types.py

```diff
@@ -1265,8 +1267,13 @@ def name(self) -> str:
         """The name as it is shown to the external world, camel-cased.
         In general, the "id" field is already camel-cased,
         but in case that didn't happen this property will fix that.
+        For nested subfields (not a relation), the name needs to be prefixed
+        for the external world.
         """
-        return toCamelCase(self._id)
+        name = self._id
+        if self._parent_field is not None and self._parent_field.is_nested_object:
+            name = f"{self._parent_field._id}_{self._id}"
+        return toCamelCase(name)
```

Comment:
Recursion for deep nesting:

```suggestion
        name = toCamelCase(self._id)
        if self._parent_field is not None and self._parent_field.is_nested_object:
            name = f"{self._parent_field.name}_{name}"  # Note: recursive use of .name.
        return name
```